src/frontend/src/optimizer/plan_node/generic.rs

```diff
@@ -293,7 +293,8 @@ impl PlanAggCall {
     pub fn partial_to_total_agg_call(&self, partial_output_idx: usize) -> PlanAggCall {
         let total_agg_kind = match &self.agg_kind {
             AggKind::Min | AggKind::Max | AggKind::StringAgg | AggKind::FirstValue => self.agg_kind,
-            AggKind::Count | AggKind::Sum | AggKind::ApproxCountDistinct => AggKind::Sum,
+            AggKind::Count | AggKind::Sum0 => AggKind::Sum0,
+            AggKind::Sum | AggKind::ApproxCountDistinct => AggKind::Sum,
```

Comment:
Should `ApproxCountDistinct` also use `Sum0` as its final phase?

Reply:
Not really sure. I think both are ok.